src/components/lib/FlowGrid/FlowGrid.jsx

```diff
@@ -176,6 +176,22 @@ export default class FlowGrid extends Component{
     return Math.max(6, lowestItem, selected) || 6;
   }
 
+  _onReturn(l, down){
+    const {row, column} = l
+    const newRow = down ? row + 1 : (row || 1) -1
+    const newLocation = {row: newRow, column}
+    this.props.onAddItem(newLocation)
```

Comment:
Do we still need to AddItem?  I feel like it's not needed when there's an item there.  Maybe there should be a combined action for these two.